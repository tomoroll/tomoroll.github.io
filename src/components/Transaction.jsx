import React from 'react';
import PropTypes from 'prop-types';
import { etherscanUrls } from '../utils/etheroll-contract';


function Transaction({ hash, network }) {
  const url = `${etherscanUrls[network]}/tx/${hash}`;
  return <a href={url}>{hash}</a>;
}
Transaction.propTypes = {
  hash: PropTypes.string.isRequired,
  network: PropTypes.number.isRequired,
};

export default Transaction;
