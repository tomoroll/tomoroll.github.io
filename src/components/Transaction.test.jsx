import React from 'react';
import renderer from 'react-test-renderer';
import Transaction from './Transaction';


it('renders correctly on empty transactions', () => {
  const tree = renderer
    .create(<Transaction hash="0x1234" network={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
