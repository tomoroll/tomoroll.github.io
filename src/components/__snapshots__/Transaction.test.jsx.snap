// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly on empty transactions 1`] = `
<a
  href="https://ropsten.etherscan.io/tx/0x1234"
>
  0x1234
</a>
`;
